src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOnSession.java

```diff
@@ -160,16 +161,22 @@ public void put(SecurityIdentity identity) {
                             connection.getInputStream().close();
                         } catch (Exception cause) {
                             log.warnHttpMechSsoFailedLogoutParticipant(remoteURI.toString(), cause);
+                            // failed to logout participant, remove it from the list of participants
+                            try (SingleSignOn target = context.getSingleSignOnManager().find(id)) {
+                                target.removeParticipant(participantId);
+                            }
                         }
                     });
 
-                    try (SingleSignOn target = this.context.getSingleSignOnManagerManager().find(id)) {
+                    try (SingleSignOn target = this.context.getSingleSignOnManager().find(id)) {
                         if (target != null) {
                             // If all logout requests were successful, then there should be no participants, and we can invalidate the SSO
-                            if (target.getParticipants().isEmpty()) {
+                            if (!target.getParticipants().isEmpty()) {
+                                log.debugf("Destroying SSO [%s]. Participant list not empty.", target.getId());
+                            } else {
                                 log.debugf("Destroying SSO [%s]. SSO is no longer associated with any participants", target.getId());
-                                target.invalidate();
                             }
+                            target.invalidate();
```

Comment:
If we're going to invalidate the SSO no matter what, we don't need to separately remove participants whose logout request failed.  So, I think we can eliminate that whole try block in the above catch clause.